homeassistant/components/enphase_envoy/number.py

```diff
@@ -0,0 +1,126 @@
+"""Number platform for Enphase Envoy solar energy monitor."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+
+from pyenphase import EnvoyDryContactSettings
+
+from homeassistant.components.number import (
+    NumberDeviceClass,
+    NumberEntity,
+    NumberEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import EnphaseUpdateCoordinator
+from .entity import EnvoyBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class EnvoyRelayRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[EnvoyDryContactSettings], float]
+
+
+@dataclass
+class EnvoyRelayNumberEntityDescription(
+    NumberEntityDescription, EnvoyRelayRequiredKeysMixin
+):
+    """Describes an Envoy Dry Contact Relay number entity."""
+
+
+RELAY_ENTITIES = (
+    EnvoyRelayNumberEntityDescription(
+        key="soc_low",
```

Comment:
Does this need a min max and step?

Reply:
Accepts 0-100 which I believe is the default min/max for `number` entities